### PR TITLE
Do not instantiate IService instances unecessarily at startup

### PR DESCRIPTION
**Pull request by mark-monteiro**
Every single class that implements `IService` is being instantiated at startup just so that we can call `getType()` on it. This PR changes that logic to just use the types directly without instantiating them.

**Changes**
* Do not instantiate the services to get their `Type`
* Create `ServiceController` via dependency injection instead of manually using the constructor
* When registering a service with `ServiceController`, log a warning if the type passed does not implement `IService`

### CODE REVIEW

File: Emby.Server.Implementations/Services/ServiceController.cs [hunk 1]
```diff
@@ -21,6 +34,13 @@ public void Init(HttpListenerHost appHost, IEnumerable<Type> serviceTypes)
 
         public void RegisterService(HttpListenerHost appHost, Type serviceType)
         {
+            // Make sure the provided type implements IService
+            if (!typeof(IService).IsAssignableFrom(serviceType))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/ServiceController.cs:38]: Why not
```suggestion
            if (!(serviceType is IService))
```
?

File: Emby.Server.Implementations/Services/ServiceController.cs [hunk 2]
```diff
@@ -3,14 +3,27 @@
...
 
 namespace Emby.Server.Implementations.Services
 {
     public delegate object ActionInvokerFn(object intance, object request);
+
     public delegate void VoidActionInvokerFn(object intance, object request);
 
     public class ServiceController
     {
+        private readonly ILogger _log;
```
Comment (Bond-009) [inline Emby.Server.Implementations/Services/ServiceController.cs:16]: Maybe rename to `_logger` to keep it consistent with the rest of the codebase?

File: Emby.Server.Implementations/Services/ServiceController.cs (hunk 1 above)
Comment (mark-monteiro) [reply Emby.Server.Implementations/Services/ServiceController.cs:38]: `serviceType` is a `Type` variable, not an `IService`

Comment (Bond-009) [reply Emby.Server.Implementations/Services/ServiceController.cs:38]: Right, my bad

File: Emby.Server.Implementations/Services/ServiceController.cs (hunk 2 above)
Comment (mark-monteiro) [reply Emby.Server.Implementations/Services/ServiceController.cs]: Done

**Pull request merged**